src/3p-analytics-common.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @enum {string} */
+export const AMP_ANALYTICS_3P_MESSAGE_TYPE = {
+  EVENT: 'E',
+  RESPONSE: 'R',
+};
+
+/** @typedef {!Object<!string,!Array<!string>>} */
+export let AmpAnalytics3pEvent;
+// Example:
+// {
+//   "2": ["viewed=true&...etc.", ... ],
+//   ...
+// }
+
+/** @typedef {JsonObject} */
+export let AmpAnalytics3pResponse;
+// Example:
+// {"status":"received","somethingElse":"42"}
+
+/**
+ * A class for holding AMP Analytics third-party vendors responses to frames.
+ */
+export class ResponseMap {
+  /**
+   * Add a response
+   * @param {!string} frameType The identifier for the third-party frame that
+   * responded
+   * @param {!string} creativeUrl The URL of the creative being responded to
```

Comment:
It comes from amp-analytics.js calling this.win.document.baseURI
Wouldn't it be entirely likely that 2 different creatives both use the same vendor? If so, we would not one a response for one to be applied to the other, right?